libcxx/include/__text_encoding/text_encoding.h

```diff
@@ -0,0 +1,1482 @@
+      {2093, "ebcdic-no-277+euro"},
+      {2094, "IBM01143"},
+      {2094, "CCSID01143"},
+      {2094, "CP01143"},
+      {2094, "csIBM01143"},
+      {2094, "ebcdic-fi-278+euro"},
+      {2094, "ebcdic-se-278+euro"},
+      {2095, "IBM01144"},
+      {2095, "CCSID01144"},
+      {2095, "CP01144"},
+      {2095, "csIBM01144"},
+      {2095, "ebcdic-it-280+euro"},
+      {2096, "IBM01145"},
+      {2096, "CCSID01145"},
+      {2096, "CP01145"},
+      {2096, "csIBM01145"},
+      {2096, "ebcdic-es-284+euro"},
+      {2097, "IBM01146"},
+      {2097, "CCSID01146"},
+      {2097, "CP01146"},
+      {2097, "csIBM01146"},
+      {2097, "ebcdic-gb-285+euro"},
+      {2098, "IBM01147"},
+      {2098, "CCSID01147"},
+      {2098, "CP01147"},
+      {2098, "csIBM01147"},
+      {2098, "ebcdic-fr-297+euro"},
+      {2099, "IBM01148"},
+      {2099, "CCSID01148"},
+      {2099, "CP01148"},
+      {2099, "csIBM01148"},
+      {2099, "ebcdic-international-500+euro"},
+      {2100, "IBM01149"},
+      {2100, "CCSID01149"},
+      {2100, "CP01149"},
+      {2100, "csIBM01149"},
+      {2100, "ebcdic-is-871+euro"},
+      {2101, "Big5-HKSCS"},
+      {2101, "csBig5HKSCS"},
+      {2102, "IBM1047"},
+      {2102, "IBM-1047"},
+      {2102, "csIBM1047"},
+      {2103, "PTCP154"},
+      {2103, "CP154"},
+      {2103, "Cyrillic-Asian"},
+      {2103, "PT154"},
+      {2103, "csPTCP154"},
+      {2104, "Amiga-1251"},
+      {2104, "Ami-1251"},
+      {2104, "Ami1251"},
+      {2104, "Amiga1251"},
+      {2104, "csAmiga1251"},
+      {2105, "KOI7-switched"},
+      {2105, "csKOI7switched"},
+      {2106, "BRF"},
+      {2106, "csBRF"},
+      {2107, "TSCII"},
+      {2107, "csTSCII"},
+      {2108, "CP51932"},
+      {2108, "csCP51932"},
+      {2109, "windows-874"},
+      {2109, "cswindows874"},
+      {2250, "windows-1250"},
+      {2250, "cswindows1250"},
+      {2251, "windows-1251"},
+      {2251, "cswindows1251"},
+      {2252, "windows-1252"},
+      {2252, "cswindows1252"},
+      {2253, "windows-1253"},
+      {2253, "cswindows1253"},
+      {2254, "windows-1254"},
+      {2254, "cswindows1254"},
+      {2255, "windows-1255"},
+      {2255, "cswindows1255"},
+      {2256, "windows-1256"},
+      {2256, "cswindows1256"},
+      {2257, "windows-1257"},
+      {2257, "cswindows1257"},
+      {2258, "windows-1258"},
+      {2258, "cswindows1258"},
+      {2259, "TIS-620"},
+      {2259, "ISO-8859-11"},
+      {2259, "csTIS620"},
+      {2260, "CP50220"},
+      {2260, "csCP50220"},
+      {0, nullptr} // sentinel
+  };
+
+  const __encoding_data* __encoding_rep_ = __text_encoding_data + 1;
+  char __name_[max_name_length + 1]      = {0};
+};
+
+template <>
+struct hash<text_encoding> {
+  size_t operator()(const text_encoding& __enc) const noexcept { return std::hash<text_encoding::id>()(__enc.mib()); }
+};
+
+namespace ranges {
+
+template <>
+inline constexpr bool enable_borrowed_range<text_encoding::aliases_view> = true;
+
+} // namespace ranges
```

Comment:
It seems better not to reopen the inner namespace.
```suggestion
template <>
inline constexpr bool ranges::enable_borrowed_range<text_encoding::aliases_view> = true;
```